include/spdlog/sinks/ringbuffer_sink-inl.h

```diff
@@ -0,0 +1,46 @@
+// Copyright(c) 2015-present, Gabi Melman & spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+
+#pragma once
+
+#ifndef SPDLOG_HEADER_ONLY
+#include "spdlog/sinks/ringbuffer_sink.h"
+#endif
+
+#include "spdlog/common.h"
+#include "spdlog/details/os.h"
+
+namespace spdlog {
+namespace sinks {
+
+template<typename Mutex>
+SPDLOG_INLINE ringbuffer_sink<Mutex>::ringbuffer_sink(size_t buf_size)
+{
+    buf_=details::circular_q<details::log_msg_buffer>(buf_size);
+}
+
+template<typename Mutex>
+SPDLOG_INLINE void ringbuffer_sink<Mutex>::sink_it_(const details::log_msg &msg)
+{
+    buf_.push_back(details::log_msg_buffer{msg});
+}
+
+template<typename Mutex>
+SPDLOG_INLINE std::vector<std::string> ringbuffer_sink<Mutex>::last(size_t lim)
+{
```

Comment:
Note this will return in a confusing order (the head is not at position 0). I think what we need here is way to iterate over the items in the actual order.

Reply:
As far as I see, you are mistaken... `circular_q::at(i)` will return item at `(head_+i)%max_items_` in the storage vector.

Reply:
Right. I didn't notice correctly.